src/main/java/ch/njol/skript/expressions/ExprPlayerWeather.java

```diff
@@ -40,35 +38,23 @@
  */
 @Name("Player Weather")
 @Description("The weather for a player.")
-@Examples({"set weather for arg-player to clear",
-		"arg-player's weather is rainy"})
+@Examples({"set weather of arg-player to rainy",
+		"reset player's weather",
+		"if arg-player's weather is rainy"})
 @Since("2.2-dev34")
-public class ExprPlayerWeather extends PropertyExpression<Player, WeatherType> {
+public class ExprPlayerWeather extends SimplePropertyExpression<Player, WeatherType> {
 	static {
-		Skript.registerExpression(ExprPlayerWeather.class, WeatherType.class, ExpressionType.PROPERTY,
-				"[the] weather [(for|of) %player%]", "%player%'[s] weather");
+		PropertyExpression.register(ExprPlayerWeather.class, WeatherType.class, "weather", "player");
```

Comment:
You don't support multiple players in the syntax yet you support it in the code.